scipy/sparse/linalg/_isolve/minres.py

```diff
@@ -153,25 +154,25 @@ def minres(A, b, x0=None, *, rtol=1e-5, shift=0.0, maxiter=None,
         x = b
         return (x, 0)
 
-    beta1 = sqrt(beta1)
+    beta1 = sqrt(beta1.real)
 
     if check:
         # are these too strict?
 
         # see if A is symmetric
         w = matvec(y)
         r2 = matvec(w)
-        s = inner(w,w)
-        t = inner(y,r2)
+        s = dotprod(w,w)
+        t = dotprod(y,r2)
         z = abs(s - t)
         epsa = (s + eps) * eps**(1.0/3.0)
         if z > epsa:
             raise ValueError('non-symmetric matrix')
 
         # see if M is symmetric
         r2 = psolve(y)
-        s = inner(y,y)
-        t = inner(r1,r2)
+        s = dotprod(y,y)
+        t = dotprod(r1,r2)
```

Comment:
note: these lines don't seem to be covered by tests at the moment when checking locally via `spin test -j 12 --coverage`

<img width="546" height="240" alt="Image" src="https://github.com/user-attachments/assets/2932d6d6-b8c3-4cb1-924e-aff99bbce476" />